arch/risc-v/src/common/riscv_fpu.S

```diff
@@ -95,7 +108,7 @@ up_fpuconfig:
  *   floating point registers.
  *
  * C Function Prototype:
- *   void riscv_savefpu(uint64_t *regs);
+ *   void riscv_savefpu(uintptr_t *regs);
```

Comment:
should we change the prototype in up_internal.h:
```
#ifdef CONFIG_ARCH_FPU
#ifdef CONFIG_ARCH_RV64
void riscv_savefpu(uint64_t *regs);
void riscv_restorefpu(const uint64_t *regs);
#else /* !CONFIG_ARCH_RV64 */
void riscv_savefpu(uint32_t *regs);
void riscv_restorefpu(const uint32_t *regs);
#endif /* CONFIG_ARCH_RV64 */
#else
#  define riscv_savefpu(regs)
#  define riscv_restorefpu(regs)
#endif
```
to:
#ifdef CONFIG_ARCH_FPU
void riscv_savefpu(uintptr_t *regs);
void riscv_restorefpu(const uintptr_t *regs);
#else
#  define riscv_savefpu(regs)
#  define riscv_restorefpu(regs)
#endif

Reply:
Yes, but it would lead to many warnings, it can be done while merging arch.h/irq.h/mcause.h/syscall.h.

Reply:
Ok, let's do in the next refactor PR.